core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java

```diff
@@ -245,31 +248,37 @@ public void printDebug() {
 	@Override
 	public Completable processAsync() {
 		return Completable.defer(() -> {
-			// Process the batch
 			Completable obs = Completable.complete();
+			// Map<Boolean, List<SearchQueueEntry>> entries2 = getEntries().stream()
+			// .collect(Collectors.partitioningBy(i -> i.getElementAction() == STORE_ACTION));
+			//
+			// List<SearchQueueEntry> storeEntries = entries2.get(true);
+			// List<SearchQueueEntry> nonStoreEntries = entries2.get(false);
+			//
+
 			List<? extends SearchQueueEntry> nonStoreEntries = getEntries().stream().filter(i -> i.getElementAction() != STORE_ACTION).collect(
-					Collectors.toList());
+				Collectors.toList());
 
 			List<? extends SearchQueueEntry> storeEntries = getEntries().stream().filter(i -> i.getElementAction() == STORE_ACTION).collect(Collectors
-					.toList());
+				.toList());
 
```

Comment:
Use .collect(Collectors.partitioning) instead of calling getEntries twice. Each getEntries sorts the list which can be costly when having a lot of elements